### PR TITLE
[T4PUB-331] Add GitHub action

### PR DESCRIPTION
**Pull request by Bodong-Yang**
Add a test workflow for each PR.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: test CI
+
+on: [pull_request]
+
+jobs:
+  pytest_with_coverage:
+    runs-on: ubuntu-latest
```
Comment (obi-t4) [inline .github/workflows/test.yml:7]: I'd like specify the specific version
```suggestion
    runs-on: ubuntu-20.04
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    timeout-minutes: 3
+    strategy:
+      matrix:
+        python-version: [3.7]
+    steps:
+      - uses: actions/checkout@v2
```
Comment (obi-t4) [inline .github/workflows/test.yml:13]: It is better to have a name:
```suggestion
      - name: checkout
        uses: actions/checkout@v2
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+on: [pull_request]
+
+jobs:
+  pytest_with_coverage:
+    runs-on: ubuntu-latest
+    timeout-minutes: 3
+    strategy:
+      matrix:
+        python-version: [3.7]
```
Comment (obi-t4) [inline .github/workflows/test.yml:11]: we're using python3.8
```suggestion
        python-version: [3.8]
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install package dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install pyyaml requests
+          for pathfile in $(find ./ -name "requirements*.txt" -type f); do
+             pip install -r $pathfile
+          done
```
Comment (obi-t4) [inline .github/workflows/test.yml:24]:
```suggestion
          python -m pip install -r app/requirements.txt
```
is enough I think.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+      # pytest
+      - name: Install test dependencies
+        run: |
+          python -m pip install pytest pytest-mock pytest-custom_exit_code pytest-cov
```
Comment (obi-t4) [inline .github/workflows/test.yml:28]: OK. I'd like to introduce requirements-dev.txt in the tests directory. It's OK, not this PR.

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Execute pytest with coverage
+        run: |
+          set -o pipefail
+          sudo $pythonLocation/bin/python -m pytest --cov-report term-missing --cov=app ./tests | tee pytest-coverage.txt
+      # export the coverage report to the comment!
+      - name: Comment coverage
+        uses: coroo/pytest-coverage-commentator@v1.0.2
+  python_lint_check:
+    runs-on: ubuntu-latest
```
Comment (obi-t4) [inline .github/workflows/test.yml:37]:
```suggestion
    runs-on: ubuntu-20.04
```

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    timeout-minutes: 3
+    strategy:
+      matrix:
+        python-version: [3.7]
```
Comment (obi-t4) [inline .github/workflows/test.yml:41]:
```suggestion
        python-version: [3.8]
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    steps:
+      - uses: actions/checkout@v2
```
Comment (obi-t4) [inline .github/workflows/test.yml:43]: It is better to have a name

**Review by obi-t4 — COMMENTED** [review]
Please check the comments.

File: .github/workflows/test.yml (hunk 4 above)
Comment (Bodong-Yang) [reply .github/workflows/test.yml:24]: Line 21: The python lib configured by the actions/setup-python@v1 do have **pyyaml** and **requests** built-in. Maybe add these 2 packages in the requirement-dev.txt in the future?

Comment (obi-t4) [reply .github/workflows/test.yml]: I see. So please keep `python -m pip install pyyaml requests`. pyyaml and requests should be add in the requiremnts.txt in the future.

File: .github/workflows/test.yml (hunk 2 above)
Comment (obi-t4) [reply .github/workflows/test.yml:14]:
```suggestion
        uses: actions/checkout@v2
```

File: .github/workflows/test.yml (hunk 8 above)
Comment (obi-t4) [reply .github/workflows/test.yml:43]:
```suggestion
        uses: actions/checkout@v2
```

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -10,6 +10,7 @@ jobs:
       matrix:
         python-version: [3.8]
     steps:
+      - name: Checkout commit
       - uses: actions/checkout@v2
```
Comment (Bodong-Yang) [inline .github/workflows/test.yml]: Oh I have a syntax error here, please let me fix it first.

File: .github/workflows/test.yml [hunk 10]
```diff
@@ -40,6 +39,7 @@ jobs:
...
```
Comment (Bodong-Yang) [inline .github/workflows/test.yml]: same as above.

File: .github/workflows/test.yml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+    runs-on: ubuntu-20.04
+    timeout-minutes: 3
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+      - name: Checkout commit
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
```
Comment (obi-t4) [inline .github/workflows/test.yml:16]: Did you try setup-pytyon@v2 ?
I'm not sure difference but current implementation is v2

Comment (Bodong-Yang) [reply .github/workflows/test.yml]: V2 version's python lib still don't have pyyaml and requests built-in.

**Review by obi-t4 — APPROVED** [review]
LGTM :+1:

**Pull request merged**